main.go

```diff
@@ -0,0 +1,362 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	mapset "github.com/deckarep/golang-set"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+)
+
+func main() {
+	processMonorailFruits()
+}
+
+func portTestExpectations() {
+	//f, err := os.Open("TestExpectations.txt")
+	f, err := os.Open("PortableTestExpectation.txt")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	//bsfSet := getBSF()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		expectation := scanner.Text()
+		url, _, testname, results := parseExpectation(expectation)
+		/*
+			if isMeetCriteria(url, tags, testname, results) {
+				if bsfSet.Contains(testname) {
+					fmt.Println(expectation)
+				}
+			}
+		*/
+		toWPTMetadata(url, testname, results)
+
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// https://github.com/web-platform-tests/wpt.fyi/issues/1993#issuecomment-654551961
+func isMeetCriteria(url, tags, testname, results string) bool {
+	if url == "" {
+		return false
+	}
+
+	if testname == "" {
+		return false
+	}
+
+	if results != "Timeout" && results != "Failure" {
+		return false
+	}
+
+	if tags != "" {
+		if !strings.Contains(tags, "Linux") {
+			if tags != "Release" {
+				return false
+			}
+		} else {
+			if strings.Contains(tags, "Debug") {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// syntax: https://chromium.googlesource.com/chromium/src/+/master/docs/testing/web_test_expectations.md#syntax
+// [ bugs ] [ "[" modifiers "]" ] test_name [ "[" expectations "]" ]
+func parseExpectation(line string) (url string, tags string, testname string, results string) {
+	if strings.HasPrefix(line, "# ") {
+		return
+	}
+
+	split := strings.Split(line, " ")
+	//fmt.Println("new line")
+	//fmt.Println(split)
+	for i := 0; i < len(split); i++ {
+		if strings.HasPrefix(split[i], "crbug.com") {
```

Comment:
it would be nice if this could cope with WebKit's test expectations as well

Reply:
Sorry for the late reply! @clopez should have a script that imports WebKit's test expectations as he has done it before for (wpt-metadata)[https://github.com/web-platform-tests/wpt-metadata/commits?author=clopez}